ompi/mpi/java/c/mpi_MPI.c

```diff
@@ -1154,6 +1153,9 @@ jboolean ompi_java_exceptionCheck(JNIEnv *env, int rc)
         (*env)->DeleteLocalRef(env, jmessage);
         return JNI_TRUE;
     }
+    else if (JNI_TRUE == jni_exception) {
+        return JNI_TRUE;
```

Comment:
From what I have seen so far as I have started changing some of the error code is that cleanup is happening in the case of any exception (MPI included), but objects are being returned to Java that should not be.  The JNI stuff gets cleaned up, but its possible there is some MPI related stuff that is not getting cleaned up.  Ill have to look more closely at that, but a first pass probably won't include those changes.

I think the non blocking MPI subroutines get their errors through the Request object that is created, but I could be wrong about that.  Its been a while since I have looked at that code.

I personally don't think we should be checking user buffers, but I suppose if someone wanted to add it as a feature they could.  So long as there was a switch to turn it on or off.  And that is probably something that would need to be discussed with more people.